Makefile

```diff
@@ -0,0 +1,59 @@
+RELEASE ?= `grep -1 -A 0 -B 0 '<version>' pom.xml | head -n 1 | awk '{print $1}' | sed -e 's/.*<version>//' | sed -e 's/<\/version>.*//'`
```

Comment:
Can you please put a comment blurb at the top about the purpose of this Makefile?  I just wanna allow for quick perusers of the repo to not be hopefully *too* overwhelmed by our extensive collection of build/setup-related scripts:
* Vagrantfile
* Dockerfile
* Makefile
* bin/build-release.sh
* pom.xml

Also, Make is a kind of painful language to handle IMHO, so I wonder if this might be easier to maintain in a different language. No need to rewrite this now, we can come back later if we decide we like something else better.